### PR TITLE
Fix granules API client list querying

### PR DESCRIPTION
**Pull request by laurenfrederick**
**Summary:** This was causing intermittent test failures

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: packages/api-client/src/executions.ts [hunk 1]
```diff
@@ -44,16 +44,18 @@ export const getExecution = async (params: {
  */
 export const getExecutions = async (params: {
   prefix: string,
+  query?: { [key: string]: string },
```
Comment (markdboyd) [inline packages/api-client/src/executions.ts:47]: Typescriptin like a boss

File: packages/api-client/tests/test-executions.js [hunk 2]
```diff
@@ -37,7 +37,8 @@ test('getExecutions calls the callback with the expected object', async (t) => {
     payload: {
       httpMethod: 'GET',
       resource: '/{proxy+}',
-      path: '/executions'
+      path: '/executions',
+      queryStringParameters: undefined
```
Comment (markdboyd) [inline packages/api-client/tests/test-executions.js:41]: Why can't we just exclude `queryStringParameters` if we're wanting it to be `undefined`?

File: packages/api-client/tests/test-granules.js [hunk 3]
```diff
@@ -228,14 +228,14 @@ test('moveGranule calls the callback with the expected object', async (t) => {
...
-  const query = 'testQuery';
+  const query = { limit: 50 };
   const expected = {
     prefix: t.context.testPrefix,
     payload: {
       httpMethod: 'GET',
       resource: '/{proxy+}',
       path: '/granules',
-      body: JSON.stringify({ query })
+      queryStringParameters: { limit: 50 }
```
Comment (markdboyd) [inline packages/api-client/tests/test-granules.js:238]:
```suggestion
      queryStringParameters: query
```

File: packages/api-client/tests/test-granules.js [hunk 4]
```diff
@@ -258,7 +258,7 @@ test('listGranules calls the callback with the expected object if there is no qu
       httpMethod: 'GET',
       resource: '/{proxy+}',
       path: '/granules',
-      body: undefined
+      queryStringParameters: undefined
```
Comment (markdboyd) [inline packages/api-client/tests/test-granules.js:261]: Same comment as above: why include this property if we don't want it to defined?

File: packages/api-client/tests/test-executions.js (hunk 2 above)
Comment (laurenfrederick) [reply packages/api-client/tests/test-executions.js:41]: The test will holler at you, since that is the expected value

**Pull request merged**